src/bench/pqs/spraylist.cpp

```diff
@@ -55,14 +55,38 @@ void
 spraylist::insert(const uint32_t &k,
                   const uint32_t &v)
 {
-    sl_add(m_q, k, READ_ADD_REM_ELASTIC_TX);
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
+}
+
+void
+spraylist::insert(const size_t &k,
+                  const size_t &v)
+{
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
 }
 
 bool
-spraylist::delete_min(uint32_t &v)
+spraylist::delete_min(size_t &k, size_t &v)
 {
-    const int ret = spray_delete_min(m_q, &v, d);
-    return (ret == 1);
+    unsigned long k_ret;
+    unsigned long v_ret;
+    int ret;
+    do {
+        ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
```

Comment:
Wow that's a pretty noticeable difference. Anyway, I'll defer to your best judgement on this since it's  been a while that I looked at all of this.

My gut reaction is that 1. it'd be nice to keep the behavior of existing benchmarks consistent (unless they're incorrect), 2. behavior of different PQs should be consistent (it seems weird if only some loop in delete_min), and 3. making the shortest path benchmark more scalable sounds good.

Let me know when you think this is ready to merge.